### DISCUSSION TITLE
How to Get All Listings/ASINS for a Given Seller

### DISCUSSION THREAD

**Discussion by nicolasheady** [amzn/selling-partner-api-models#3537] [General]
Is there a best way to get a list of all ASINs for a seller? I though I could use "searchCatalogItems" and include as queries: the seller's ID for "sellerId" and for "keywords" the character "*". However, it doesn't seem that inputting the seller ID actually restricts the search to only that seller.

It appears like the only other way is using the Reports API to generate a "GET_MERCHANT_LISTINGS_ALL_DATA" report, then download it.

Is there another way?

**Comment by drewxa** [comment]
> Is there another way?

Unfortunately not. although SP API tries to seem like a RESTful and do everything in accordance with the best practices of the REST API, there is no such operation as getting all listings yet. but it would be very useful

What about searchCatalogItems? This is just a full-text search for products, which carries little useful signal and does not have any additional filtering functions. The details in this comment https://github.com/amzn/selling-partner-api-docs/discussions/2017#discussioncomment-2127666
